eip/EIPS/eip-4337.md

```diff
@@ -462,24 +430,10 @@ it is critical that the exact same opcode and precompile banning rules as well a
 for the `handleOps` validation in its entirety as for individual UserOperations.
 Otherwise, attackers may be able to use the banned opcodes to detect running on-chain and trigger a `FailedOp` revert.
 
-Banning an offending entity for a given bundler is achieved by increasing its `opsSeen` value by `1000000`
-and removing all UserOperations for this entity already present in the mempool.
-This change will allow the negative reputation value to deteriorate over time consistent with other banning reasons.
-
-If any of the three conditions is violated, the client should reject the `op`. If both calls succeed (or, if `op.paymaster == ZERO_ADDRESS` and the first call succeeds)without violating the three conditions, the client should accept the op. On a bundler node, the storage keys accessed by both calls must be saved as the `accessList` of the `UserOperation`
+The storage slots accessed during the above tracing may be used as "accessList" when submitting the transaction.
```

Comment:
Couple things about this section:

The above is suggesting to use a `debug_traceCall` on the entire `handleOps` transaction. Shouldn't a bundler only need to care about the validation section of `handleOps`? Tracing the entire thing seems like a waste of resources.

> use the consumed gas for the actual transaction execution. 

This seems like an easy attack on a bundler: have something use less gas during simulation than during onchain execution. To me, the only safe way to do this is to use the gas limit of each UO (plus a buffer) as the total gas limit of the. 

I understand that we eventually want bundlers to be block builder such that they know the exact onchain state when they run the simulation. However I'm not sure we should be baking that into the procedures suggested here.

> The storage slots accessed during the above tracing may be used as "accessList" when submitting the transaction.

Similarly to above, the slots accessed during execution can change between bundle simulation and execution. It seems only safe to use slots accessed during validation in the access list. Again, that changes if you know the exact state and have guaranteed top of block inclusion, but I don't see why we should be baking this into the spec.

Reply:
2. I agree that the gas rules should be removed, and replaced with general suggestions: They are not really required by the "consensus", and should be defined by the bundler itself.

Reply:
3. re: accessList building - you're right. We indeed specified it as may (which doesn't imply preference), but it can (may?/should?) be removed....

Reply:
Agreed with the above. I feel like (3) should be removed unless its really pressing.

(will split into individual comments next time, sorry about the numbering)